sdk/table/azure-table/azure/table/_table_client.py

```diff
@@ -0,0 +1,509 @@
+        :param table_name: The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Any
+        :returns: A table client.
+        :rtype: ~azure.table.TableClient
+        """
+        account_url, secondary, credential = parse_connection_str(
+            conn_str, credential, 'queue')
+        if 'secondary_hostname' not in kwargs:
+            kwargs['secondary_hostname'] = secondary
+        return cls(account_url, table_name=table_name, credential=credential, **kwargs)  # type: ignore
+
+    @classmethod
+    def from_table_url(cls, table_url, credential=None, **kwargs):
+        # type: (str, Optional[Any], Any) -> TableClient
+        """A client to interact with a specific Table.
+
+        :param str table_url: The full URI to the table, including SAS token if used.
+        :param Any credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token. The value can be a SAS token string, an account
+            shared access key, or an instance of a TokenCredentials class from azure.identity.
+        :returns: A table client.
+        :rtype: ~azure.table.TableClient
+        """
+        try:
+            if not table_url.lower().startswith('http'):
+                table_url = "https://" + table_url
+        except AttributeError:
+            raise ValueError("Table URL must be a string.")
+        parsed_url = urlparse(table_url.rstrip('/'))
+
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL: {}".format(table_url))
+
+        table_path = parsed_url.path.lstrip('/').split('/')
+        account_path = ""
+        if len(table_path) > 1:
+            account_path = "/" + "/".join(table_path[:-1])
+        account_url = "{}://{}{}?{}".format(
+            parsed_url.scheme,
+            parsed_url.netloc.rstrip('/'),
+            account_path,
+            parsed_url.query)
+        table_name = unquote(table_path[-1])
+        if not table_name:
+            raise ValueError("Invalid URL. Please provide a URL with a valid queue name")
+        return cls(account_url, table_name=table_name, credential=credential, **kwargs)
+
+    @distributed_trace
+    def get_table_access_policy(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> List
+        """Retrieves details about any stored access policies specified on the table that may be
+        used with Shared Access Signatures.
+
+                :keyword callable cls: A custom type or function that will be passed the direct response
```

Comment:
indentation should line up with the start of the docstring.
also we don't need to document `cls`